sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -2063,30 +2112,26 @@ object RegexRewriteUtils {
    */
   def matchSimplePattern(ast: RegexAST): RegexOptimizationType = {
     ast.children() match {
-      case (RegexChar('^') | RegexEscaped('A')) :: RegexGroup(_, RegexSequence(parts), None) :: rest
-          if isliteralString(parts) && rest.forall(isWildcard) => {
-        // ^(literal).* => startsWith literal
-        RegexOptimizationType.StartsWith(RegexCharsToString(parts))
-      }
       case (RegexChar('^') | RegexEscaped('A')) :: ast 
           if isliteralString(stripTailingWildcards(ast)) => {
         // ^literal.* => startsWith literal
+        println("Starts with optimization")
         RegexOptimizationType.StartsWith(RegexCharsToString(stripTailingWildcards(ast)))
       }
-      case noStartsWithAst => stripLeadingWildcards(noStartsWithAst) match {
-        case RegexGroup(_, RegexSequence(parts), None) :: rest
-            if isliteralString(parts) && rest.forall(isWildcard) => {
-          // (literal).* => contains literal
-          RegexOptimizationType.Contains(RegexCharsToString(parts))
-        }
-        case ast if isliteralString(stripTailingWildcards(ast)) => {
-          // literal.* => contains literal
-          RegexOptimizationType.Contains(RegexCharsToString(stripTailingWildcards(ast)))
-        }
-        case _ => {
+      case astLs => {
+        val noStartsWithAst = stripTailingWildcards(stripLeadingWildcards(astLs))
+        val prefixRangeInfo = getPrefixRangePattern(noStartsWithAst)
+        if (prefixRangeInfo.isDefined) {
+          val (prefix, length, start, end) = prefixRangeInfo.get
```

Comment:
{x,y} is not supported with this. It is just {x}. Please update the PR to either support it, or so it is clear in all of the documentation that it is not supported, and that we fallback if we see an {x,y}

Reply:
`{x,y}` is supported and `y` is useless in the `rlike` case, so it is omitted here, we only need to pass `x` into kernel to support this case.